pass-journal-loader/pass-journal-loader-nih/src/main/java/org/eclipse/pass/loader/journal/nih/BatchJournalFinder.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2017 Johns Hopkins University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.pass.loader.journal.nih;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.pass.support.client.PassClient;
+import org.eclipse.pass.support.client.PassClientSelector;
+import org.eclipse.pass.support.client.model.Journal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Analyzes journals in our repository in order to match incoming journals against our existing journals
+ *
+ * @author apb@jhu.edu
+ */
+public class BatchJournalFinder implements JournalFinder {
+
+    private Logger LOG = LoggerFactory.getLogger(BatchJournalFinder.class);
+
+    private Map<String, Set<String>> issnMap = new HashMap<>();
+
+    private Map<String, Set<String>> nlmtaMap = new HashMap<>();
+
+    private Map<String, Set<String>> nameMap = new HashMap<>();
+
+    private Set<String> foundUris = new HashSet<>();
+
+    private void load(PassClient client) throws IOException {
+        LOG.info("Loading existing journals from PASS");
+
+        AtomicInteger count = new AtomicInteger(0);
+
+        PassClientSelector<Journal> sel = new PassClientSelector<>(Journal.class);
+        client.streamObjects(sel).forEach(j -> {
+
+            count.incrementAndGet();
+
+            j.getIssns().forEach(issn -> {
+                update(issnMap, issn, j.getId());
+            });
+
+            update(nlmtaMap, j.getNlmta(), j.getId());
+            update(nameMap, j.getJournalName(), j.getId());
+        });
+
+        LOG.info("Loaded " + count.get() + " existing journals");
+    }
+
+    private static void update(Map<String, Set<String>> map, String key, String value) {
```

Comment:
Looks like `static` can be removed.

Reply:
That was deliberate, it just indicates the method doesn't mutate instance variables, but doesn't really matter either way/